### ISSUE TITLE
DTE-IG results and bootstrapping

### ISSUE THREAD

**Issue by neryah** [vicliv/DTE#1]
Could you please clarify the bootstrapping process?
Did you randomly bootstrap just once for each dataset (as in the GitHub code), or did you perform multiple bootstraps per dataset?
[How can I get the DTE-IG results?]

From the paper:
“For the unsupervised setting, we sample the whole dataset with replacement for the training data, while the test data is the whole dataset. This bootstrapping method allows us to test the variance over the training dataset for each method.”

**Comment by vicliv** [comment]
Yes, we did the bootstrap as it is in GitHub, and reported the results over five runs (five different seeds). For DTE-IG, I believe the hyperparameters are not the same as the ones we used in our experiments reported in the paper. The exact set of hyperparameters was unfortunately lost, but here are the parameters that should give you close results:

`def
__init__(self,
seed, 
model_name="diffusion_gamma",
hidden_size=[128, 128],
epochs=500,
batch_size=64,
lr=1e-4,
weight_decay=0,
T=300,
num_bins=0):`

**Comment by neryah** [comment]
Thanks. What AUCROC did you get with this __init__(...) hyperparameters for the inverse-gamma model? I assume the results depend on the run order, but I got 47.21 in semi-supervised settings and 47.4 in unsupervised (average across all datasets).

**Comment by vicliv** [comment]
I did not run it entirely, but it should definitely be higher than this; similar to the results from the paper.

**Comment by neryah** [comment]
[aggregated_orig_semi_AUCROC.csv](https://github.com/user-attachments/files/17003349/aggregated_orig_semi_AUCROC.csv)

Here are the results across all 5 seeds.

P.S. for the categorical model, the AUCROC is much closer to what’s reported in your paper.

**Comment by vicliv** [comment]
Hi, sorry for the delay. I was investigating the issue. There was a missing square when predicting the anomaly score for that method. The predict_score method is now specific to the DTEInverseGamma class. Sorry for the inconvenience, I believe this issue appeared when combining all DTE methods into the same file.
